### PR TITLE
Additions to the workflow for each benchmark.

### PR DESCRIPTION
**Pull request by mattmccutchen-cci**
- Convert with -alltypes as well as without.

- Compile both converted programs, ignoring failure for the -alltypes version (we can still view the compile errors in the logs).

Other improvements:

- Since the workflow file contains so much repetitive code that the workflow format does not support factoring out, the file is now generated by a Python program, `generate-workflow.py`.  Since it is easy to forget to re-run `generate-workflow.py` after changing it, the workflow stops at the beginning if it detects it is out of date with `generate-workflow.py`.

- For consistency, use the Checked C compiler for the pre-conversion build of each benchmark as well as the post-conversion build. To avoid regressing the ability to convert vsftpd at all, a fix for the error that blocked vsftpd from building with the Checked C compiler even before conversion is included in this PR.
 
- Improve some names and clean up some shell commands.

- Try not to overload gamera.

### CODE REVIEW

**Comment by mattmccutchen-cci** [comment]
Is it time to start using a script to generate the workflow file?  Maybe in Python to be consistent with most of our other scripts, even if there might be more specialized languages that are arguably better suited to the task?  Writing more short scripts to factor out command sequences (in this repository if we don't want to put them in the 3C repository) would help a little, but there's no need to do that if we generate the whole workflow file with a script anyway.

**Comment by mwhicks1** [comment]
This makes a lot of sense to me. I think what you are saying is that different targets require different sequences of steps to make them amenable for conversion, and sometimes only particular conversion steps should be carried out. Thus, we should write a meta description that indicates these steps, and a single script to generate the actions file that executes them. Is that right?

**Comment by mattmccutchen-cci** [comment]
> I think what you are saying is that different targets require different sequences of steps to make them amenable for conversion, and sometimes only particular conversion steps should be carried out.

Correct.

> Thus, we should write a meta description that indicates these steps, and a single script to generate the actions file that executes them. Is that right?

Roughly.  I don't think our scale is large enough yet that it makes sense to have a formal "meta description" separate from the script, though of course the whole script could be viewed as a meta description.

**Comment by mattmccutchen-cci** [comment]
[The new run](https://github.com/correctcomputation/actions/actions/runs/625926927) looks at least as good as the previous run, so I'm declaring this ready for review.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -62,127 +67,358 @@ jobs:
             -DLLVM_USE_SPLIT_DWARF=ON \
             -DLLVM_ENABLE_PROJECTS="clang" \
             ${{github.workspace}}/depsfolder/checkedc-clang/llvm
-          ninja 3c clang
+          # -l 36: Try not to overload gamera. Hopefully this will use all 36
+          # hyperthreads when nothing else is running and automatically scale
+          # back when other jobs are running. TODO: Better solution?
+          ninja -l 36 3c clang
```
[the remark this answers is not in the document]
Comment (mwhicks1) [reply .github/workflows/main.yml:73]: OK, sounds good (making it the default practice on gamera).

[the remark this answers is not in the document]
Comment (mattmccutchen-cci) [reply .github/workflows/main.yml:83]: Resolved for this PR.  I've made a note elsewhere to follow up on documenting `-l 36` as best practice for gamera.

**Review by john-h-kastner — APPROVED** [review]
Looks good. Lets merge this so it's easier to test our fixes against it.

One suggestion. The python script could include a call to [yamllint](https://github.com/adrienverge/yamllint) to sanity check the generated workflow. Easy enough to run manually, but it might catch trivial errors in future changes.

**Comment by mattmccutchen-cci** [comment]
> One suggestion. The python script could include a call to [yamllint](https://github.com/adrienverge/yamllint) to sanity check the generated workflow. Easy enough to run manually, but it might catch trivial errors in future changes.

A good idea but I don't want to take the time to do it now, so I've filed #9 for it.

**Pull request merged**